src/webhook.js

```diff
@@ -0,0 +1,121 @@
+import axiosRetry, { exponentialDelay, linearDelay } from "axios-retry";
+import { HttpsProxyAgent } from "https-proxy-agent";
+import Config from "./config.js";
+import SlackError from "./errors.js";
+
+/**
+ * This Webhook class posts the configured payload to the provided webhook, with
+ * whatever additional settings set.
+ */
+export default class Webhook {
+  /**
+   * @param {Config} config
+   */
+  async post(config) {
+    if (!config.inputs.webhook) {
+      throw new SlackError(config.core, "No webhook was provided to post to");
+    }
+    /**
+     * @type {import("axios-retry").IAxiosRetryConfig}
+     * @see {@link https://www.npmjs.com/package/axios-retry}
+     */
+    const retries = this.retries(config.inputs.retries);
+    axiosRetry(config.axios, retries);
+    await config.axios
+      .post(config.inputs.webhook, config.content.values, {
+        ...this.proxies(config),
+      })
+      .then((response) => {
```

Comment:
Mixing async/await with promise usage in here can be confusing. Can we consolidate around a single approach?

Reply:
Changed this to use `async` and `await` here and elsewhere! Thanks for catching this confusion :pray: ✨